examples/medplum-demo-bots/src/deduplication/merge-bot.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  BotEvent,
+  createReference,
+  deepClone,
+  getCodeBySystem,
+  getQuestionnaireAnswers,
+  MedplumClient,
+} from '@medplum/core';
+import {
+  Identifier,
+  Patient,
+  QuestionnaireResponse,
+  QuestionnaireResponseItemAnswer,
+  Reference,
+  Resource,
+  RiskAssessment,
+} from '@medplum/fhirtypes';
+
+interface MergedPatients {
+  readonly src: Patient;
+  readonly target: Patient;
+}
+
+interface Subject {
+  subject?: Reference<Patient>;
+}
+
+type ResourceWithSubject = Resource & Subject;
+
+/**
+ * Links two patient records indicating one replaces the other.
+ *
+ * @param src - The source patient record which is being replaced.
+ * @param target - The target patient record which will replace the source.
+ * @returns - Object containing updated source and target patient records with their links.
+ */
+export function linkPatientRecords(src: Patient, target: Patient): MergedPatients {
+  const targetLinks = target.link ?? [];
+  targetLinks.push({ other: createReference(src), type: 'replaces' });
+  const targetCopy = deepClone(target);
+
+  const srcLinks = src.link ?? [];
+  srcLinks.push({ other: createReference(target), type: 'replaced-by' });
+  const srcCopy = deepClone(src);
+  return { src: { ...srcCopy, link: srcLinks, active: false }, target: { ...targetCopy, link: targetLinks } };
+}
+
+/**
+ * Merges contact information (identifiers) of two patient records.
+ * Identifiers from the source are marked as 'old' in the merged record.
```

Comment:
Would put a link to the spec for $merge, as justification for _why_ we might do that